optics-core/src/Optics/Monoidal.hs

```diff
@@ -0,0 +1,193 @@
+  -- no AfterJoin with         A_ReversedLens
+  AfterJoin An_AffineFold      A_ReversedPrism    = A_Fold
+  AfterJoin An_AffineFold      A_Prism            = A_Fold
+  -- no AfterJoin with         A_Review
+  AfterJoin An_AffineFold      A_Lens             = A_Fold
+  AfterJoin An_AffineFold      A_Getter           = A_Fold
+  AfterJoin An_AffineFold      An_AffineTraversal = A_Fold
+  AfterJoin An_AffineFold      A_Traversal        = A_Fold
+  AfterJoin An_AffineFold      A_Fold             = A_Fold
+  -- no AfterJoin with         A_Setter
+
+  -- A_Traversal-----
+  AfterJoin A_Traversal        An_Iso             = A_Traversal
+  -- no AfterJoin with         A_ReversedLens
+  AfterJoin A_Traversal        A_ReversedPrism    = A_Fold
+  AfterJoin A_Traversal        A_Prism            = A_Traversal
+  -- no AfterJoin with         A_Review
+  AfterJoin A_Traversal        A_Lens             = A_Traversal
+  AfterJoin A_Traversal        A_Getter           = A_Fold
+  AfterJoin A_Traversal        An_AffineTraversal = A_Traversal
+  AfterJoin A_Traversal        An_AffineFold      = A_Fold
+  AfterJoin A_Traversal        A_Fold             = A_Fold
+  AfterJoin A_Traversal        A_Setter           = A_Setter
+
+  -- A_Fold-----
+  AfterJoin A_Fold             An_Iso             = A_Fold
+  -- no AfterJoin with         A_ReversedLens
+  AfterJoin A_Fold             A_ReversedPrism    = A_Fold
+  AfterJoin A_Fold             A_Prism            = A_Fold
+  -- no AfterJoin with         A_Review
+  AfterJoin A_Fold             A_Lens             = A_Fold
+  AfterJoin A_Fold             A_Getter           = A_Fold
+  AfterJoin A_Fold             An_AffineTraversal = A_Fold
+  AfterJoin A_Fold             An_AffineFold      = A_Fold
+  AfterJoin A_Fold             A_Traversal        = A_Fold
+  -- no AfterJoin with         A_Setter
+
+  -- no AfterJoin with A_Setter-----
+  -- A_Traversal-----
+  AfterJoin A_Setter           An_Iso             = A_Setter
+  -- no AfterJoin with         A_ReversedLens
+  -- no AfterJoin with         A_ReversedPrism
+  AfterJoin A_Setter           A_Prism            = A_Setter
+  -- no AfterJoin with         A_Review
+  AfterJoin A_Setter           A_Lens             = A_Setter
+  -- no AfterJoin with         A_Getter
+  AfterJoin A_Setter           An_AffineTraversal = A_Setter
+  -- no AfterJoin with         An_AffineFold
+  -- no AfterJoin with         A_Fold
+
+  -- END GENERATED CONTENT
+
+  -- Every optic kinds can be joined with itself.
+  AfterJoin k k = k
+
+  -- Everything else is a type error.
+  AfterJoin k l = TypeError ('ShowType k
+                       ':<>: 'Text " and "
+                       ':<>: 'ShowType l
+                       ':<>: 'Text " cannot be composed sequentially")
+
+infixr 6 <++>
+class After is m where
+  nothing :: Optic' m is s a
+  (<++>)  :: (Is k m, Is l m, m ~ AfterJoin k l)
+          => Optic' k is s a -> Optic' l is s a
+          -> Optic' m is s a
```

Comment:
In general, we tend to avoid introducing new operators in `optics`, or at least provide a non-operator equivalent. Is there precedent for this operator elsewhere? I suppose `summing` would be the obvious name choice if we don't mind a slightly breaking change.

Reply:
I didn't know about that rule. I haven't seen this operator used anywhere else, but I think that `summing` would work great here (and would provide a good backwards-compatibility story for those already using `summing` in `Fold`).